### PR TITLE
New Basemaps: Portal and Raster Categories

### PR DESCRIPTION
**Pull request by jenmerritt**
### Description

This PR is for the following sample categories:
- Portal
- Raster

Notes specific to this PR: 
- Identify Raster Cell: indentation was tidied up

### General comments on Basemap updates:

It updates `Map` and/or `Basemap` Constructors as per the new design. This may include the following changes depending on the sample:
- Replace `Basemap.Type` with `BasemapStyle`
- If a lat/long/zoom was being set in the `Map` constructor, this is replaced with a `Viewpoint` being set on the `MapView`.
- Any Readme Updates identified

In addition, the samples that include the above changes, or other services requiring an API Key going forward, have the API Key  set within the sample. This will then work in conjunction with the Gradle Task being added in PR 593.

Once 593 is approved and merged into `dev` we will merge those changes into these PRs and then update all the Samples to `100.10.0-xxxx` and do a round of testing of the Samples with all the API Key / Basemap changes at that time. (We have left them as 100.9.0 (in most cases) for now to minimize merge conflicts in the `build.gradle` file).

In the meantime, the updates listed above can be reviewed. Thanks!

### CODE REVIEW

File: portal/oauth/src/main/java/com/esri/samples/oauth/OAuthSample.java
Comment (JonLavi) [inline portal/oauth/src/main/java/com/esri/samples/oauth/OAuthSample.java]:
```suggestion
      // create a map with the imagery basemap style
```

File: raster/colormap-renderer/src/main/java/com/esri/samples/colormap_renderer/ColormapRendererSample.java
Comment (JonLavi) [inline raster/colormap-renderer/src/main/java/com/esri/samples/colormap_renderer/ColormapRendererSample.java]: I realise I'm commenting on these to change to `the`, but Im not 100% sure it's correct. My rationale is since it's a unique value from an enum, there can't be any other `imagery basemap style`s, so it's correct to refer to as `the style` rather than `a style`?
```suggestion
      // create a map with the imagery basemap style
```

File: raster/identify-raster-cell/src/main/java/com/esri/samples/identify_raster_cell/IdentifyRasterCellSample.java
Comment (JonLavi) [inline raster/identify-raster-cell/src/main/java/com/esri/samples/identify_raster_cell/IdentifyRasterCellSample.java]:
```suggestion
      // create a map with the oceans basemap style
```

File: raster/raster-function/src/main/java/com/esri/samples/raster_function/RasterFunctionSample.java
Comment (JonLavi) [inline raster/raster-function/src/main/java/com/esri/samples/raster_function/RasterFunctionSample.java]: I won't comment on any other instances, before I make a fool of myself!
```suggestion
      // create a map with the dark gray basemap style
```

File: raster/raster-layer-url/src/main/java/com/esri/samples/raster_layer_url/RasterLayerURLSample.java
Comment (JonLavi) [inline raster/raster-layer-url/src/main/java/com/esri/samples/raster_layer_url/RasterLayerURLSample.java]: Just said I won't comment, but this one definitely needs either `a` or `the`, doesn't matter which 😁 
```suggestion
      // create a map with a dark gray basemap style
```

**Review by JonLavi — APPROVED** [review]
LGTM @jenmerritt , sorry if I'm causing confusion with my comments on the choice between `a`/`the` basemap style . I won't comment on those in the remainder of the PRs, assuming you would have picked the correct one before they are merged.

File: raster/colormap-renderer/src/main/java/com/esri/samples/colormap_renderer/ColormapRendererSample.java
Comment (jenmerritt) [reply raster/colormap-renderer/src/main/java/com/esri/samples/colormap_renderer/ColormapRendererSample.java]: Yeah I think 'a basemap style' but 'the specific basemap style' sounds correct.

File: portal/oauth/README.md
Comment (Rachael-E) [inline portal/oauth/README.md]: is the BasemapStyle.ARCGIS_STREETS basemap still classed as a public basemap layer? I wasn't sure if it's accessible without an API key and therefore if it's ok to still refer to it as public. To be on the safe side we could just go with:

> otherwise the map will contain only the basemap style layer

**Review by Rachael-E — APPROVED** [review]
just a query that might be worth checking: if the map is truly public (and doesn't need an api key to access it) then leave as is, otherwise we should change. I think all 2D maps need api keys now though.

File: portal/oauth/README.md
Comment (jenmerritt) [reply portal/oauth/README.md]: Nice spot! That sounds right. I'll update here for now, but I think these will also be revisited across the designs at some point in the future too 👍

**Comment by jenmerritt** [comment]
`dev` merged in (including API Key gradle tasks changes). Samples that have an API Key set have been tested running from the IDE.

Await merging these changes until release.

**Pull request merged**
